frontend/src/components/category-stats.tsx

```diff
@@ -0,0 +1,317 @@
+                        
+                        return (
+                          <Box key={index}>
+                            <Text size="sm" style={{ color: entry.color }}>
+                              {courseCode}: {Number(meanMark).toFixed(1)}%
+                            </Text>
+                            {organiser && (
+                              <Text size="xs" c="dimmed">
+                                CO: <Text span style={{ color: courseOrganiserColors[organiser]?.replace('0.3', '1.0') || 'inherit' }}>
+                                  {organiser}
+                                </Text>
+                              </Text>
+                            )}
+                            {stdDev && (
+                              <Text size="xs" c="dimmed">
+                                Standard deviation: ±{stdDev}%
+                              </Text>
+                            )}
+                          </Box>
+                        );
+                      })}
+                    </Paper>
+                  );
+                }
+                return null;
+              },
+            }}
+          />
+        </Paper>
+        
+        {/* Course Organiser Information */}
+        {Object.keys(courseOrganiserColors).length > 0 && (
+          <Box mt="md">
+            <Text size="sm" fw={500} mb="xs">Course Organisers Found:</Text>
+            <Group gap="md">
+              {Object.entries(courseOrganiserColors).map(([organiser, color]) => (
+                <Group key={organiser} gap="xs">
+                  <Box
+                    w={16}
+                    h={16}
+                    style={{
+                      backgroundColor: color.replace('0.3', '0.8'),
+                      borderRadius: 3,
+                    }}
+                  />
+                  <Text size="xs">{organiser}</Text>
+                </Group>
+              ))}
+            </Group>
+            <Text size="xs" c="dimmed" mt="xs">
+              All course organisers found in the data. Hover on different chart points to see which organiser ran each course in specific years - organisers change over time!
+            </Text>
+          </Box>
+        )}
+      </Box>
+
+      {stats && stats.length > 0 && (
+        <Box>
+          <Title order={3} mb="md">
+            Course Overview
+          </Title>
+          <Group gap="md">
+            {[...new Set(stats.map(s => s.course_code))].sort().map(code => {
+              const courseStats = stats.filter(s => s.course_code === code);
+              const latestStat = courseStats
+                .sort((a, b) => a.academic_year.localeCompare(b.academic_year))
+                .pop();
+              
+              return (
+                <Paper key={code} withBorder p="md" style={{ minWidth: 200 }}>
+                  <Stack gap="xs">
+                    <Text fw={600} size="sm">{code}</Text>
```

Comment:
Text here is way too small, I suggest something like:

- code: `lg`
- name + organiser, `sm`
- mean + stddev, `md` or `sm`
- length, `xs`

If you make the Stack component's gap zero, it might look a bit more compact and better. (pending some manual adjustment to padding, maybe "xs" between the CO and mean.)